manifests/init.pp

```diff
@@ -2,9 +2,18 @@
 #
 # A description of what this class does
 #
+# @param enable_net_process_log
+#   Enable/disable the net_process_log logging
+#
 # @example
 #   include profile_audit
-class profile_audit {
+class profile_audit (
+  Boolean $enable_net_process_log,
+) {
+
+  if $enable_net_process_log {
+    include ::profile_audit::net_process_log
+  }
```

Comment:
I would normally put this parameter and logic into the subclass. I am open to being talked out of this, but I think it reads cleaner (both from reviewing the parameters and the classes).

So `net_process_log.pp` would look something like the following:
```
class profile_audit::net_process_log (
  Boolean $enable,
) {

  if $enable {
    ## THE EXISTING LOGIC WOULD BE INSERTED HERE
    ## ...
  }
} 
```

Reply:
Yeah I originally had it like that but flip flopped since it just felt strange to include something I didn't intend to use, but the point about being able to have it clean up after itself is the better behavior. Will flip it back and add some cleanup ability.

Reply:
Ok, turning the class on/off has been moved to the subclass